cognite/neat/rules/transformers/_mapping.py

```diff
@@ -95,3 +97,37 @@ def transform(self, rules: DMSRules | OutRules[DMSRules]) -> JustRules[DMSRules]
                 prop.reference = ReferenceEntity.from_entity(ref_prop.view, ref_prop.view_property)
 
         return JustRules(solution)
+
+
+class RuleMapper(RulesTransformer[InformationRules, InformationRules]):
+    """Maps properties and classes using the given mapping.
+
+    **Note**: This transformer mutates the input rules.
+
+    Args:
+        mapping: The mapping to use.
+
+    """
+
+    def __init__(self, mapping: RuleMapping) -> None:
+        self.mapping = mapping
+
+    def transform(self, rules: InformationRules | OutRules[InformationRules]) -> JustRules[InformationRules]:
+        input_rules = self._to_rules(rules)
+
+        destination_by_source = self.mapping.properties.as_destination_by_source()
+        destination_cls_by_source = self.mapping.classes.as_destination_by_source()
+        for prop in input_rules.properties:
+            if destination_prop := destination_by_source.get(prop.as_reference()):
+                prop.class_ = destination_prop.class_
+                prop.property_ = destination_prop.property_
```

Comment:
would at all be useful to add information to prop.comment ?

Reply:
Then we lose information that is already there. Or we start to clutter it. Suggest we think a bit carefully through how to do the lineage here.